src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -0,0 +1,314 @@
+
+    /**
+     * The options map.
+     */
+    private Map<String, Object> opts;
+    private ServerContext serverContext;
+
+    private static final String PREFIX_LAYOUT = "M_LAYOUT";
+    private static final String KEY_LAYOUT = "M_CURRENT";
+
+    private CorfuRuntime corfuRuntime;
+    /**
+     * Policy to be used to detect failures.
+     */
+    private IFailureDetectorPolicy failureDetectorPolicy;
+    /**
+     * Latest layout received from bootstrap or the runtime.
+     */
+    private volatile Layout latestLayout;
+    /**
+     * Interval in executing the failure detection policy.
+     * In milliseconds.
+     */
+    @Getter
+    private long policyExecuteInterval = 1000;
+    /**
+     * To schedule failure detection.
+     */
+    @Getter
+    private final ScheduledExecutorService failureDetectorService;
+    /**
+     * Future for periodic failure detection task.
+     */
+    private Future failureDetectorFuture = null;
+
+    public ManagementServer(ServerContext serverContext) {
+
+        this.opts = serverContext.getServerConfig();
+        this.serverContext = serverContext;
+
+        // Restoring state of last layout if present.
+        safeUpdateLayout(getCurrentLayout());
+        this.failureDetectorPolicy = serverContext.getFailureDetectorPolicy();
+        this.failureDetectorService = Executors.newScheduledThreadPool(
+                2,
+                new ThreadFactoryBuilder()
+                        .setDaemon(true)
+                        .setNameFormat("FaultDetector-%d-" + getLocalEndpoint())
+                        .build());
+
+        // Initiating periodic task to poll for failures.
+        try {
+            failureDetectorService.scheduleAtFixedRate(
+                    this::failureDetectorScheduler,
+                    0,
+                    policyExecuteInterval,
+                    TimeUnit.MILLISECONDS);
+        } catch (RejectedExecutionException err) {
+            log.error("Error scheduling failure detection task, {}", err);
+        }
+    }
+
+    /**
+     * Resetting the datastore entries.
+     */
+    @Override
+    public void reset() {
+        String d = serverContext.getDataStore().getLogDir();
+        if (d != null) {
+            Path dir = FileSystems.getDefault().getPath(d);
+            String prefixes[] = new String[]{PREFIX_LAYOUT, KEY_LAYOUT};
+
+            for (String pfx : prefixes) {
+                try (DirectoryStream<Path> stream = Files.newDirectoryStream(dir, pfx + "_*")) {
+                    for (Path entry : stream) {
+                        Files.delete(entry);
+                    }
+                } catch (IOException e) {
+                    log.error("reset: error deleting prefix " + pfx + ": " + e.toString());
+                }
+            }
+        }
+    }
+
+    @Override
+    public void reboot() {
+    }
+
+    @Override
+    public synchronized void handleMessage(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
```

Comment:
Please change this to use the @ServerHandler annotation (see LogUnitServer for details)